### ISSUE TITLE
Unknown DW_AT_const_value form

### ISSUE THREAD

**Issue by brenns10** [osandov/drgn#374]
We encountered `Exception: unknown DW_AT_const_value form` while reading a variable out of a stack frame. This was on a customer vmcore, so we can't share that, but the kernel version was our `4.14.35-2047.510.5.5.el7uek.x86_64`. I'm happy to do some back & forth to get any helpful supporting info.

This reminds me of #233, though "const_value" sounds at least... promising.

**Comment by osandov** [comment]
Huh, unfortunate that I didn't include what the unknown form is. Here's a patch to try out for now:

```diff
diff --git a/libdrgn/dwarf_info.c b/libdrgn/dwarf_info.c
index 549988a9..49109533 100644
--- a/libdrgn/dwarf_info.c
+++ b/libdrgn/dwarf_info.c
@@ -4534,8 +4534,15 @@ drgn_object_from_dwarf_constant(struct drgn_debug_info *dbinfo, Dwarf_Die *die,
 		}
 		return drgn_object_set_unsigned_internal(ret, &type, uvalue);
 	} else {
-		return drgn_error_create(DRGN_ERROR_OTHER,
-					 "unknown DW_AT_const_value form");
+		char *name;
+		err = drgn_format_type_name(qualified_type, &name);
+		if (err)
+			return err;
+		err = drgn_error_format(DRGN_ERROR_OTHER,
+					"unknown DW_AT_const_value form 0x%x for type %s",
+					attr->form, name);
+		free(name);
+		return err;
 	}
 }
 
```

(I'll add something similar to that upstream.)

**Comment by brenns10** [comment]
You know what? This was on our older 0.0.24 installation. I setup your patch on 0.0.25, and it didn't reproduce. So I tried 0.0.25 without the patch, and it also didn't reproduce. I guess this is "magically fixed". My guess is that the upgraded libdwfl version fixed it? I could patch 0.0.24 to see what this was, but given that it's already low-ish priority and fixed, I wouldn't pursue it unless you want.
